webhook.go

```diff
@@ -38,12 +81,36 @@ type MetadataMatcherConfig struct {
 // a webhook registration request.  The only difference between this struct and
 // the Webhook struct is the Duration field.
 type Registration struct {
+	// CanonicalName is the canonical name of the registration request.
+	// Reusing a CanonicalName will override the configurations set in that previous
+	// registration request with the same CanonicalName.
+	CanonicalName string `json:"canonical_name"`
+
 	// Address is the subscription request origin HTTP Address.
 	Address string `json:"registered_from_address"`
 
-	// Config contains data to inform how events are delivered.
+	// Deprecated: This field should only be used for backwards compatibility
+	// matching. Use ConfigWebhooks instead.
+	// Config contains data to inform how events are delivered to single url.
 	Config DeliveryConfig `json:"config"`
```

Comment:
legacy will not use new features (hashing, etc)
create a new `Registration ` for  legacy (a separate legacy into it's own `v1` api)